needy/__main__.py

```diff
@@ -196,12 +206,91 @@ def development_mode(args=[]):
             return 0 if needy.development_mode(parameters.library) else 1
         needy.set_development_mode(parameters.library, not parameters.disable)
 
+    __invoke_with_configured_needy(__development_mode, parameters)
+
+    return 0
+
+
+def cache_path(args=[]):
+    parser = argparse.ArgumentParser(
+        prog='%s cache-path' % os.path.basename(sys.argv[0]),
+        description='Configures a cache for satisfied needs.',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""available commands:
+  set    sets the cache path
+  unset  unsets the cache path
+  query  emits the current cache path
+
+Use '%s cache-path <command> --help' to get help for a specific command.
+""" % os.path.basename(sys.argv[0])
+    )
+    parser.add_argument('command', help='see below')
+    parser.add_argument('args', nargs=argparse.REMAINDER)
+    parameters = parser.parse_args(args)
+
+    def set_cache_path(args=[]):
+        needs_dir = Needy.resolve_needs_directory('.')
+        default_cache_path = os.path.join(needs_dir, '.cache') if needs_dir else None
+        parser = argparse.ArgumentParser(
+            prog='%s cache-path set' % os.path.basename(sys.argv[0]),
+            description='set a cache path.',
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument(
+            'path',
+            nargs='?',
+            default=default_cache_path,
+            help='new cache path')
+        parameters = parser.parse_args(args)
+
+        def __set_cache_path(needy):
+            configuration = {
+                'path': os.path.realpath(parameters.path),
+            }
+            needy.set_cache(DirectoryCache(configuration))
+
+        __invoke_with_configured_needy(__set_cache_path, parameters)
+
+    def unset_cache_path(args=[]):
+        parser = argparse.ArgumentParser(
+            prog='%s cache-path unset' % os.path.basename(sys.argv[0]),
+            description='unsets the cache path.',
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+        parameters = parser.parse_args(args)
+        __invoke_with_configured_needy(lambda needy: needy.set_cache(None), parameters)
+
+    def query_cache_path(args=[]):
+        parser = argparse.ArgumentParser(
+            prog='%s cache-path query' % os.path.basename(sys.argv[0]),
+            description='query the cache path.',
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+        parameters = parser.parse_args(args)
+
+        def __query_cache_path(needy):
+            cache = needy.cache()
+            if cache and cache.description():
+                print(cache.description())
+
+        __invoke_with_configured_needy(__query_cache_path, parameters)
+
+    commands = {
+        'set': set_cache_path,
+        'unset': unset_cache_path,
+        'query': query_cache_path,
+    }
+
+    if parameters.command in commands:
+        return commands[parameters.command](parameters.args)
```

Comment:
Dont think I realized how these return values should have worked. It's almost to the point where we should separate these different commands into their own files...